regression/jbmc-strings/cprover/CProverString.java

```diff
@@ -8,6 +8,29 @@ public static char charAt(String s, int i) {
             return '\u0000';
     }
 
+    public static int codePointAt(String s, int index) {
```

Comment:
How is this file used?

Reply:
It is providing alternatives to the standard Java String functions, that jbmc knows how to interpret and solve efficiently. For instance CProverString.substring will be interpreted correctly and effeciently, whereas String.substring will be incorrect if we don't provide the String.class file and quite slow if we do provide it. The ultimate goal is to have String models that call CProverString methods so that verification is efficient and the user does not have to change method calls from String to CProverString.

The file CProverString.java is necessary for compilation, but is ignored by jbmc.
The implementation of the functions normally correspond to the internal modelling in jbmc.